fmskill/model/factory.py

```diff
@@ -42,38 +46,45 @@ class ModelResult:
     >>> mr_item = ModelResult("ThirdParty.nc", item="WL", itemInfo=mikeio.EUMType.Water_Level)
     """
 
-    def __new__(self, input, *args, **kwargs):
-        import xarray as xr
-        import mikeio
+    def __new__(
+        self,
+        data: Union[
+            str, mikeio.DataArray, pd.DataFrame, pd.Series, xr.Dataset, xr.DataArray
+        ],
+        *args,
+        **kwargs,
+    ):
 
-        if isinstance(input, str):
-            filename = input
-            ext = os.path.splitext(filename)[-1]
+        if isinstance(data, str):
```

Comment:
I think it would be great if we could accept Path also :-)

Reply:
Yes I wanted to bring that up. I personally much prefer pathlib to os.path and would love to use it everywhere. But if we don't want to always cast to string, we would also need to allow for a Path object in mikeio.Open

Reply:
mikeio already supports Path everywhere - FMskill is just lacking behind

Reply:
It's been on the [longlist](https://github.com/DHI/fmskill/discussions/132) for a long time